tidb server connection panic occurs frequently with configuration "enable-forwarding = true"

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、config "enable-forwarding = true"
2、run workload
3、inject network partition between two AZ

### 2. What did you expect to see? (Required)
no  connection panic

### 3. What did you see instead (Required)
![image](https://user-images.githubusercontent.com/84712107/202078662-8c521d7a-e79f-488b-a500-b849b7ba3d69.png)

[2022/11/15 18:18:29.252 +00:00] [ERROR] [client_batch.go:594] [batchRecvLoop] [r="\"send on closed channel\""] [stack="[github.com/tikv/client-go/v2/internal/client.(*batchCommandsClient).batchRecvLoop.func1\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.3-0.20221108030801-9c0835c80eba/internal/client/client_batch.go:596\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.chansend\n\t/usr/local/go/src/runtime/chan.go:206\nruntime.chansend1\n\t/usr/local/go/src/runtime/chan.go:145\ngithub.com/tikv/client-go/v2/internal/client.(*batchCommandsClient).batchRecvLoop\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.3-0.20221108030801-9c0835c80eba/internal/client/client_batch.go:641](http://github.com/tikv/client-go/v2/internal/client.(*batchCommandsClient).batchRecvLoop.func1/n/t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.3-0.20221108030801-9c0835c80eba/internal/client/client_batch.go:596/nruntime.gopanic/n/t/usr/local/go/src/runtime/panic.go:884/nruntime.chansend/n/t/usr/local/go/src/runtime/chan.go:206/nruntime.chansend1/n/t/usr/local/go/src/runtime/chan.go:145/ngithub.com/tikv/client-go/v2/internal/client.(*batchCommandsClient).batchRecvLoop/n/t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.3-0.20221108030801-9c0835c80eba/internal/client/client_batch.go:641)"]
[2022/11/15 18:18:29.252 +00:00] [INFO] [client_batch.go:597] ["restart batchRecvLoop"]

### 4. What is your TiDB version? (Required)

./tidb-server -V
 Release Version: v6.5.0-alpha
Edition: Community
Git Commit Hash: 6af4e4b160dd5c04d7379caf2a16a3daeb9a3e56
Git Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time: 2022-11-15 11:15:45
GoVersion: go1.19.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false

--- a/DEPS.bzl
+++ b/DEPS.bzl
@@ -7145,13 +7145,13 @@ def go_deps():
         name = "com_github_tikv_client_go_v2",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/tikv/client-go/v2",
-        sha256 = "86c2da8180c318c8258d4759fcae926a5613f5ff929dfb3461ce307c15cc44a5",
-        strip_prefix = "github.com/tikv/client-go/v2@v2.0.8-0.20240322070737-05aaba6cc6f7",
+        sha256 = "aa70eaf7e7f5723a6dc0989203f5afae9c13d0162bafefcaba308be39821b96a",
+        strip_prefix = "github.com/tikv/client-go/v2@v2.0.8-0.20240402032621-125a140034d7",
         urls = [
-            "http://bazel-cache.pingcap.net:8080/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240322070737-05aaba6cc6f7.zip",
-            "http://ats.apps.svc/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240322070737-05aaba6cc6f7.zip",
-            "https://cache.hawkingrei.com/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240322070737-05aaba6cc6f7.zip",
-            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240322070737-05aaba6cc6f7.zip",
+            "http://bazel-cache.pingcap.net:8080/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240402032621-125a140034d7.zip",
+            "http://ats.apps.svc/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240402032621-125a140034d7.zip",
+            "https://cache.hawkingrei.com/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240402032621-125a140034d7.zip",
+            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240402032621-125a140034d7.zip",
         ],
     )
     go_repository(

--- a/go.mod
+++ b/go.mod
@@ -107,7 +107,7 @@ require (
 	github.com/tdakkota/asciicheck v0.2.0
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
 	github.com/tidwall/btree v1.7.0
-	github.com/tikv/client-go/v2 v2.0.8-0.20240322070737-05aaba6cc6f7
+	github.com/tikv/client-go/v2 v2.0.8-0.20240402032621-125a140034d7
 	github.com/tikv/pd/client v0.0.0-20240322051414-fb9e2d561b6e
 	github.com/timakin/bodyclose v0.0.0-20240125160201-f835fa56326a
 	github.com/twmb/murmur3 v1.1.6

